zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -78,6 +83,58 @@ use input_selection::{
     GreedyInputSelector, GreedyInputSelectorError, InputSelector, InputSelectorError,
 };
 
+/// Represents the identifier for an account that was derived from a ZIP-32 HD seed and account index.
+#[derive(Debug, Clone)]
+pub struct HDSeedAccount(
+    pub HDSeedFingerprint,
+    pub zip32::AccountId,
+    pub UnifiedFullViewingKey,
+);
+
+/// Represents an arbitrary account for which the seed and ZIP-32 account ID are not known
+/// and may not have been involved in creating this account.
+#[derive(Debug, Clone)]
+pub enum ImportedAccount {
+    /// An account that was imported via its full viewing key.
+    Full(UnifiedFullViewingKey),
+}
+
+/// The inputs for adding an account to a wallet.
+#[derive(Debug, Clone)]
+pub enum Account {
+    /// Inputs for a ZIP-32 HD account.
+    Zip32(HDSeedAccount),
+    /// Inputs for an imported account.
+    Imported(ImportedAccount),
+}
```

Comment:
(Adapted from my previous review of this in #1202.)

Ditto my previous comment: the enum documentation is inaccurate because the type does not cover the inputs necessary to add an account to a wallet (it is missing at a minimum the account birthday, and for the current `WalletWrite::create_account` method it is also missing the seed and unnecessarily including the UFVK).

I am unconvinced that `Account` should be an enum. It is not an input to anywhere (not even `WalletWrite::create_account`, despite its current documentation), and there is no documentation to give me insight into how it might be used in future, so there is no clear rationale for why it needs to be transparently constructable / matchable (or constructable at all externally) instead of being an opaque struct. An opaque struct would be much more easily extensible in future (using a private internal enum if necessary), and if the kind of account is required to be matchable somewhere, that should be handled as a separate simple public enum `AccountKind` (similar to the relationship between `std::io::Error` and `std::io::ErrorKind`).

If instead it is intended that callers will eventually manually construct an enum case to pass to `WalletWrite::create_account`, then `Account` is the wrong name for this enum; it should be named something like `AccountKind` or `AccountRoot`, be modified to include the other necessary components for creating an account, and be modified to not contain anything unnecessary (so `AccountKind::Zip32` would not contain a `ufvk` field, because that is insufficient to ensure that the UFVK belongs to the same HD seed as the other ZIP 32 accounts in the wallet).

Reply:
I anticipate this enum being the means to acquire the keys for scanning blocks. Today, that is primarily based on UFVK but that type shouldn't be what's used in all the APIs because not all account types have a UFVK.

Reply:
> I anticipate this enum being the means to acquire the keys for scanning blocks

This does not match the updated documentation on `Account`:
> /// Describes an account in terms of its UVK or ZIP-32 origins.

My previous comment remains unresolved:

> there is no documentation to give me insight into how it might be used in future, so there is no clear rationale for why it needs to be transparently constructable / matchable (or constructable at all externally) instead of being an opaque struct.

In any case, we already have `WalletRead::get_unified_full_viewing_keys` for scanning, and _if_ we add UIVK support to the `WalletRead` and `WalletWrite` traits (which I separately disagree with, because UIVKs cannot track balance, and a wallet by definition holds a balance of funds), we would alter or replace that method.

Reply:
I removed the API.